### PR TITLE
Add sample for HTTP Location Header

### PR DESCRIPTION
**Pull request by hhariri**
Add a sample for HTTP Location Header which shows how to set the header manually, as well as defining and using an extension function.

### CODE REVIEW

File: generic/samples/location-header/build.gradle [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+repositories {
+    jcenter()
+    maven { url "https://dl.bintray.com/kotlin/ktor" }
+    maven { url "https://dl.bintray.com/kotlin/kotlinx" } // Required since 0.9.4 since ktor-client-gson includes ktor-client-json that depends on kotlinx-serialization
+}
+
+dependencies {
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
```
Comment (Stexxe) [inline generic/samples/location-header/build.gradle:28]: Do you need all of those dependencies?

File: generic/samples/location-header/build.gradle [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+sourceSets {
+    main.kotlin.srcDirs = ['src']
+    test.kotlin.srcDirs = ['test']
+    main.resources.srcDirs = ['resources']
+}
+
+repositories {
+    jcenter()
+    maven { url "https://dl.bintray.com/kotlin/ktor" }
```
Comment (Stexxe) [inline generic/samples/location-header/build.gradle:23]: Do you need these maven repositories?

File: generic/samples/location-header/build.gradle [hunk 3]
```diff
@@ -0,0 +1,79 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    }
+}
+
+apply plugin: 'kotlin'
```
Comment (Stexxe) [inline generic/samples/location-header/build.gradle:10]: Why don't you use [plugins DSL](https://docs.gradle.org/current/userguide/plugins.html)?

File: generic/samples/location-header/src/LocationHeaderApplication.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+
+/**
+ * Main entrypoint of the executable that starts a Netty webserver at port 8080
+ * and registers the [module].
+ *
+ * This is a hello-world application, while the important part is that the build.gradle
```
Comment (Stexxe) [inline generic/samples/location-header/src/LocationHeaderApplication.kt:15]: Is this a hello-world application?

File: generic/samples/location-header/src/LocationHeaderApplication.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ */
+fun Application.module() {
+    routing {
+        post("/manual") {
+            call.response.header("Location", "/manual/AF4GH")
+            call.response.status(HttpStatusCode.Created)
+            call.respondText("Manually setting the location header")
+        }
+        post("/extension") {
+            call.response.created("AF4GH")
```
Comment (Stexxe) [inline generic/samples/location-header/src/LocationHeaderApplication.kt:33]: What is `AF4GH`?

**Review by Stexxe — CHANGES_REQUESTED** [review]
No link to the issue in PR's description.

File: generic/samples/location-header/src/LocationHeaderApplication.kt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * This is a hello-world application, while the important part is that the build.gradle
+ * includes all the available artifacts and serves to use as module for a scratch or to autocomplete APIs.
+ */
+fun main(args: Array<String>) {
+    embeddedServer(Netty, port = 8080) { module()}.start(wait = true)
+}
+
+/**
+ * Module that just registers the root path / and replies with a text.
```
Comment (Stexxe) [inline generic/samples/location-header/src/LocationHeaderApplication.kt:23]: This is wrong.

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: generic/samples/location-header/build.gradle (hunk 3 above)
Comment (hhariri) [reply generic/samples/location-header/build.gradle:10]: All samples will need to be updated to this, but not in this PR.

File: generic/samples/location-header/src/LocationHeaderApplication.kt (hunk 5 above)
Comment (hhariri) [reply generic/samples/location-header/src/LocationHeaderApplication.kt:34]: Added comment to indicate it's just a sample code.

File: generic/samples/location-header/src/LocationHeaderApplication.kt (hunk 4 above)
Comment (hhariri) [reply generic/samples/location-header/src/LocationHeaderApplication.kt]: Removed.

File: generic/samples/location-header/src/LocationHeaderApplication.kt (hunk 6 above)
Comment (hhariri) [reply generic/samples/location-header/src/LocationHeaderApplication.kt:22]: Removed.

File: generic/samples/location-header/build.gradle (hunk 2 above)
Comment (hhariri) [reply generic/samples/location-header/build.gradle]: Removed

File: generic/samples/location-header/build.gradle (hunk 1 above)
Comment (hhariri) [reply generic/samples/location-header/build.gradle:26]: Removed.

File: generic/samples/location-header/src/LocationHeaderApplication.kt (hunk 6 above)
Comment (Stexxe) [reply generic/samples/location-header/src/LocationHeaderApplication.kt:22]: Comment `Module that just registers the root path / and replies with a text.` is still here

**Review by Stexxe — CHANGES_REQUESTED** [review]

**Review by Stexxe — APPROVED** [review]

**Pull request merged**